game-app/game-headless/src/main/java/org/triplea/game/server/HeadlessGameServer.java

```diff
@@ -32,57 +20,58 @@
 import org.triplea.game.startup.SetupModel;
 import org.triplea.java.Interruptibles;
 import org.triplea.java.ThreadRunner;
-import org.triplea.util.ExitStatus;
 
 /** A way of hosting a game, but headless. */
 @Slf4j
 public class HeadlessGameServer {
-  public static final String BOT_GAME_HOST_COMMENT = "automated_host";
-  public static final String BOT_GAME_HOST_NAME_PREFIX = "Bot";
   private static HeadlessGameServer instance = null;
 
   private final InstalledMapsListing availableGames = InstalledMapsListing.parseMapFiles();
   private final GameSelectorModel gameSelectorModel = new GameSelectorModel();
   private final HeadlessServerSetupModel setupPanelModel =
-      new HeadlessServerSetupModel(gameSelectorModel);
+      new HeadlessServerSetupModel(gameSelectorModel, this);
   private ServerGame game = null;
   private boolean shutDown = false;
 
-  private HeadlessGameServer() {
+  private HeadlessGameServer() {}
+
+  public static void initializeHeadlessGameServerInstance() {
+    Preconditions.checkState(
+        GameRunner.headless(), "TripleA must be headless to invoke this method!");
     if (instance != null) {
       throw new IllegalStateException("Instance already exists");
     }
-    instance = this;
-
+    instance = new HeadlessGameServer();
     Runtime.getRuntime()
         .addShutdownHook(
             new Thread(
                 () -> {
                   log.info("Running ShutdownHook.");
-                  shutDown = true;
-                  Optional.ofNullable(game).ifPresent(ServerGame::stopGame);
-                  Optional.ofNullable(setupPanelModel.getPanel())
+                  instance.shutDown = true;
+                  Optional.ofNullable(instance.game).ifPresent(ServerGame::stopGame);
+                  Optional.ofNullable(instance.setupPanelModel.getPanel())
                       .ifPresent(HeadlessServerSetup::cancel);
                 }));
 
     ThreadRunner.runInNewThread(
         () -> {
           log.info("Headless Start");
-          setupPanelModel.showSelectType();
-          log.info("Waiting for users to connect.");
-          waitForUsersHeadless();
+          instance.setupPanelModel.showSelectType();
+          instance.waitForUsersHeadless();
         });
   }
 
+  /**
+   * Returns the singleton instance of the HeadlessGameServer object.
+   *
+   * @deprecated Do not use this method, pass the instance along call trees to avoid static
+   *     coupling.
+   */
+  @Deprecated
   public static synchronized HeadlessGameServer getInstance() {
     return instance;
   }
```

Comment:
Unfortunately I was unable to remove the last usage of this method because it's being used by the ChatAppender for client logging. If youknow howlogback works, feel free to remove it, but I don't get how the API works.